Documentation/ExtensionArchitecture/HowTo/BackendModule/CreateModule.rst

```diff
@@ -0,0 +1,159 @@
+
+When creating a controller without Extbase an instance of :php:`ModuleTemplate` is required
+to return the rendered template:
+
+.. code-block:: php
+
+   use TYPO3\CMS\Backend\Template\ModuleTemplateFactory;
+   use TYPO3\CMS\Core\Imaging\IconFactory;
+
+   class AdminModuleController
+   {
+       public function __construct(
+           protected readonly ModuleTemplateFactory $moduleTemplateFactory,
+           protected readonly IconFactory $iconFactory,
+           // ...
+       ) {
+       }
+   }
+
+The controller needs to be registered in the :file:`Configuration/Services.yaml`
+with the tag `backend.controller` so that dependency injection works:
+
+.. code-block:: yaml
+   :caption: EXT:examples/Configuration/Services.yaml
+
+   services:
+      _defaults:
+         autowire: true
+         autoconfigure: true
+         public: false
+
+      T3docs\Examples\:
+         resource: '../Classes/*'
+         exclude: '../Classes/Domain/Model/*'
+
+      T3docs\Examples\Controller\AdminModuleController:
+         tags: ['backend.controller']
+
+
+Main entry point
+================
+
+:php:`handleRequest()` method is the main entry point which triggers only the allowed actions.
+This makes it possible to include e.g. Javascript for all actions in the controller.
+
+.. code-block:: php
+   :caption: EXT:examples/Classes/Controller/AdminModuleController.php
+
+   public function handleRequest(ServerRequestInterface $request): ResponseInterface
+   {
+       $languageService = $this->getLanguageService();
+       $languageService->includeLLFile('EXT:examples/Resources/Private/Language/AdminModule/locallang.xlf');
+
+       $this->menuConfig($request);
+       $moduleTemplate = $this->moduleTemplateFactory->create($request, 't3docs/examples');
+       $this->setUpDocHeader($moduleTemplate);
+
+       $title = $languageService->sL('LLL:EXT:examples/Resources/Private/Language/AdminModule/locallang_mod.xlf:mlang_tabs_tab');
+       switch ($this->MOD_SETTINGS['function']) {
+           case 'debug':
+               $moduleTemplate->setTitle($title, $languageService->getLL('module.menu.debug'));
+               return $this->debugAction($moduleTemplate);
+           case 'password':
+               $moduleTemplate->setTitle($title, $languageService->getLL('module.menu.password'));
+               return $this->passwordAction($moduleTemplate);
+           default:
+               $moduleTemplate->setTitle($title, $languageService->getLL('module.menu.log'));
+               return $this->logAction($moduleTemplate);
+       }
+   }
+
+Actions
+=======
+
+Now create an example :php:`indexAction()` and assign variables to your view
+as you would normally do.
+
+.. code-block:: php
+   :caption: EXT:examples/Classes/Controller/AdminModuleController.php
+
+   public function debugAction(
+       ModuleTemplate $view,
+       string $cmd = 'cookies'
+   ): ResponseInterface
+   {
+       $cmd = $_POST['tx_examples_admin_examples']['cmd'];
```

Comment:
Please don't use $_POST: We have the PSR-7 request object in handleRequest(), but anyway, the `$cmd` and the following switch is superfluous.